src/Estimators/EstimatorManagerNew.cpp

```diff
@@ -31,25 +32,103 @@
 #include "hdf/hdf_archive.h"
 #include "OhmmsData/AttributeSet.h"
 #include "Estimators/CSEnergyEstimator.h"
+#include "type_traits/variant_help.hpp"
 
 //leave it for serialization debug
 //#define DEBUG_ESTIMATOR_ARCHIVE
 
 namespace qmcplusplus
 {
-//initialize the name of the primary estimator
-EstimatorManagerNew::EstimatorManagerNew(const QMCHamiltonian& ham,  Communicate* c)
+
+EstimatorManagerNew::EstimatorManagerNew(const QMCHamiltonian& ham, Communicate* c)
     : RecordCount(0), my_comm_(c), max4ascii(8), FieldWidth(20)
 {
-  app_log() << "  Adding a default LocalEnergyEstimator for the MainEstimator " << std::endl;
+  app_log() << " Legacy constructor adding a default LocalEnergyEstimator for the MainEstimator " << std::endl;
   max4ascii = ham.sizeOfObservables() + 3;
-  addMainEstimator(std::make_unique<LocalEnergyEstimator>(ham, true));  
+  addMainEstimator(std::make_unique<LocalEnergyEstimator>(ham, true));
 }
 
-EstimatorManagerNew::~EstimatorManagerNew()
+template<class EstInputType, typename VAR, typename... Args>
+bool EstimatorManagerNew::createEstimator(VAR& input, Args&&... args)
 {
+  if (has<EstInputType>(input))
+  {
+    operator_ests_.push_back(std::make_unique<typename EstInputType::Consumer>(std::move(std::get<EstInputType>(input)),
+                                                                               std::forward<Args>(args)...));
+    app_log() << "EstimatorManager-> Adding " << operator_ests_.back()->get_my_name();
+    return true;
+  }
+  else
+    return false;
+}
+
+template<class EstInputType, typename VAR, typename... Args>
+bool EstimatorManagerNew::createScalarEstimator(VAR& input, Args&&... args)
+{
+  if (has<EstInputType>(input))
+  {
+    auto estimator = std::make_unique<typename EstInputType::Consumer>(std::move(std::get<EstInputType>(input)),
+                                                                       std::forward<Args>(args)...);
+    if (estimator->isMainEstimator())
+      addMainEstimator(std::move(estimator));
+    else
+      scalar_ests_.push_back(std::move(estimator));
+    return true;
+  }
+  else
+    return false;
+}
+
+using SPOMap = std::map<std::string, const std::unique_ptr<const SPOSet>>;
+
+//initialize the name of the primary estimator
+EstimatorManagerNew::EstimatorManagerNew(Communicate* c,
+                                         EstimatorManagerInput&& emi,
```

Comment:
With the current use of emi, I think we can pass it by value.
Then EstimatorManagerNew can be constructed by passing std::move(merged_input).
The following to while loops can be just range-based loops without the need of pop_back().
emi gets automatically destroyed when the constructor ends.

Reply:
I think that's a good simplification at this point.  Will address

Reply:
I think this one is missed.

Reply:
This would be the merged input being moved into EstimatorManagerNew, we can make this just a copy as well but I thought the plan was to leave it.